### PR TITLE
Workflows Execution Engine with conditional evaluation

### PR DESCRIPTION
**Pull request by PawelPeczek-Roboflow**
# Description

Next level of functionalities for Workflows EE - with true condition evaluation in-place

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update (not delivered now, postponed)

## How has this change been tested, please provide a testcase or example of how you tested the change?

YOUR_ANSWER

* CI still green
* deployed for staging, so far so good 👍 

For example, documentation changes, usability, usage/costs, secrets, etc.

## Docs

-   [ ] Docs updated? What were the changes:

### CODE REVIEW

File: inference/core/interfaces/http/orjson_utils.py
Comment (yeldarby) [inline inference/core/interfaces/http/orjson_utils.py:114]: Is this just because we don't use it anymore?

File: inference/core/interfaces/stream/inference_pipeline.py
Comment (yeldarby) [inline inference/core/interfaces/stream/inference_pipeline.py:425]: Why experimental? Seems fairly core

```suggestion
        reason="Usage of workflows with `InferencePipeline` is an experimental feature. Please report any issues "
```

File: inference/core/workflows/core_steps/common/query_language/entities/enums.py
Comment (yeldarby) [inline inference/core/workflows/core_steps/common/query_language/entities/enums.py:50]: top_most, bottom_most, center_most?

Would this also be where largest, smallest would go?

File: inference/core/workflows/core_steps/flow_control/continue_if.py
Comment (yeldarby) [inline inference/core/workflows/core_steps/flow_control/continue_if.py:41]: Would old `Condition` blocks still work if we left them in here (but deprecated & hidden from UI)?

File: inference/core/workflows/core_steps/models/foundation/clip_comparison.py
Comment (yeldarby) [inline inference/core/workflows/core_steps/models/foundation/clip_comparison.py:200]: Should we register these as outputs? Or handle context/benchmarking in a different way?

File: inference/core/interfaces/stream/inference_pipeline.py
Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/stream/inference_pipeline.py:425]: 👍

File: inference/core/workflows/core_steps/common/query_language/entities/enums.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/common/query_language/entities/enums.py:50]: I guess so - I am open to discuss UQL in general, as this is very initial draft, I do not have strong opinions yet on lots of topics related to this.

File: inference/core/workflows/core_steps/flow_control/continue_if.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/flow_control/continue_if.py:41]: We could do optional else statement - remaining Condition functionalities

File: inference/core/workflows/core_steps/models/foundation/clip_comparison.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/models/foundation/clip_comparison.py:200]: I guess we should not, as those are mainly `Nones` in practice. Apart from that - similarly to `sv.Detections(...)` change, this problem will be gone when we build better version of step - encapsulating output into something more well designed

File: inference/core/workflows/core_steps/flow_control/continue_if.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/flow_control/continue_if.py:41]: @hansent @EmilyGavrilenko @grzegorz-roboflow - please state ur opinions on how we shape condition block

File: inference/core/interfaces/http/orjson_utils.py
Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/http/orjson_utils.py:114]: yes

Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/http/orjson_utils.py:114]: I just dont want to put it into thrash without notice to people using it

File: development/stream_interface/workflows_demo.py
Comment (hansent) [inline development/stream_interface/workflows_demo.py:19]: noting interface change, do we need to update examples / code snippets anyhere?  I think in app code snippets all use named workflow so are ok.

File: inference/core/interfaces/stream/model_handlers/workflows.py
Comment (hansent) [inline inference/core/interfaces/stream/model_handlers/workflows.py:31]: woah, this means we get parallel video processing on same workflow via batch?

File: inference/core/version.py
Comment (hansent) [inline inference/core/version.py:1]: will be 0.13 for release?

File: inference/core/workflows/core_steps/flow_control/continue_if.py
Comment (hansent) [reply inference/core/workflows/core_steps/flow_control/continue_if.py:41]: If `Condition` can have optional (or just default `no-op`/ `stop-execution` value), I have slight preference for Condition with `if` and `else` as opposed to just `ContinueIf`.  I think having both would be OK.   

I think just having `ContinueIf` is fine for shipping this version and we can add additional block with else later if users ask for it / are confused by ContinueIf / constantly run into wanting an else and having to define two blocks for it.

File: inference/core/workflows/core_steps/models/foundation/lmm_classifier.py
Comment (hansent) [inline inference/core/workflows/core_steps/models/foundation/lmm_classifier.py:93]: [not a blocker] could this be inferred by type of parameters on run method of class?

File: inference/core/workflows/core_steps/models/roboflow/instance_segmentation.py
Comment (hansent) [inline inference/core/workflows/core_steps/models/roboflow/instance_segmentation.py:307]: ⚔️ 🐛

File: inference/core/workflows/core_steps/models/roboflow/object_detection.py
Comment (hansent) [inline inference/core/workflows/core_steps/models/roboflow/object_detection.py:275]: wut?  this one was definitely working before right?  does it work with the classification url also and we have the different url just in case it changes?

File: inference/core/workflows/core_steps/sinks/roboflow/roboflow_dataset_upload.py
Comment (hansent) [inline inference/core/workflows/core_steps/sinks/roboflow/roboflow_dataset_upload.py:230]: this also really makes building custom blocks much easier to get into I think!

File: inference/core/workflows/core_steps/models/roboflow/object_detection.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/models/roboflow/object_detection.py:275]: yeah, it makes no difference almost, but prefer to bring it to order

File: development/stream_interface/workflows_demo.py
Comment (PawelPeczek-Roboflow) [reply development/stream_interface/workflows_demo.py:19]: if code snippets provide examples how to deal with result we may need to change them

File: inference/core/interfaces/stream/model_handlers/workflows.py
Comment (PawelPeczek-Roboflow) [reply inference/core/interfaces/stream/model_handlers/workflows.py:31]: not everything will work (like Roboflow sink due to asyncio), but yes

File: inference/core/version.py
Comment (PawelPeczek-Roboflow) [reply inference/core/version.py:1]: yeah, will change that once merged into `development` branch or just before release

File: inference/core/workflows/core_steps/fusion/detections_consensus.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/fusion/detections_consensus.py:188]: 👍

File: inference/core/workflows/core_steps/models/foundation/lmm_classifier.py
Comment (PawelPeczek-Roboflow) [reply inference/core/workflows/core_steps/models/foundation/lmm_classifier.py:93]: Depends I guess:
* the fact that step is SIMD (takes batch oriented data and applies the same processing) is detected and handled without this method being used
* for SIMD steps we use this method to see if block expects elements of the batch to be unpacked by EE or not - inferring that from signature may be a fragile code and my hunch is that we should not be doing it, as the standard practice for people may be to implement `run(...)` without type hints which are optional and in Python only small pool of tools (including for instance Pydantic) analyses type hints annotations as first order of business

**Review by hansent — APPROVED** [review]
approving merge into dev branch after testing session

**Pull request merged**
